lessons/basics/typespec.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: page
+title: Specifications and types
+category: basics
```

Comment:
I think we should classify this as an "advanced" lesson.  You don't _need_ to have specs and types to work with Elixir, it comes after you've grasped the basics.